### PR TITLE
use day js

### PR DESCRIPTION
**Pull request by mktcode**
Continues work from: https://github.com/snapshot-labs/snapshot/pull/972.

### CODE REVIEW

File: crowdin.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 files:
-  - source: /src/locales/default.json
+  - source: /src/locales/en-US.json
```
Comment (bonustrack) [inline crowdin.yml:2]: Is it possible to not change this?

Comment (bonustrack) [inline crowdin.yml:2]: Is it possible to not change this?

Comment (mktcode) [reply crowdin.yml:2]: I would have asked you how much of an impact that actually has. I can revert this, sure. :D

File: src/views/SpaceCreate.vue [hunk 2]
```diff
@@ -379,15 +380,13 @@ watchEffect(() => {
           </UiButton>
           <UiButton
             @click="(modalOpen = true), (selectedDate = 'start')"
-            :disabled="props.space.voting?.delay"
```
Comment (samuveth) [inline src/views/SpaceCreate.vue:382]: This shouldn't be removed.

File: src/views/SpaceCreate.vue [hunk 3]
```diff
@@ -379,15 +380,13 @@ watchEffect(() => {
...
             class="w-full mb-2"
           >
             <span v-if="!dateStart">{{ $t('create.startDate') }}</span>
             <span v-else v-text="$d(dateStart * 1e3, 'short', 'en-US')" />
           </UiButton>
           <UiButton
             @click="(modalOpen = true), (selectedDate = 'end')"
-            :disabled="props.space.voting?.period"
```
Comment (samuveth) [inline src/views/SpaceCreate.vue:390]: This shouldn't be removed.

**Comment by mktcode** [comment]
I worked around [the issue with Vite and dynamic imports from packages](https://github.com/vitejs/vite/issues/772) by using a symlink, so that Vite thinks the day.js locales are "local" project files. This way, after building, everything works fine and as expected and day.js locales are requested by the client on demand.

The next issue is reactivity. Changing language in the current implementation will not update the UI. I'll have to take a closer look at Vue's reactivity system but since this PR already sits here for a while, I might go for a "naive" solution.

**Comment by mktcode** [comment]
Ok... I thought I had a workaround for [Vite's dynamic import issue](https://github.com/vitejs/vite/issues/772) but I don't. I tried using a symlink to fool Vite into thinking it's handling "local" project files and that works but causes other problems.

And then there is the reactivity. In the current implementation, UI doesn't update when switching the language. And implementing that on top of a semi-optimal workaround doesn't make a lot of sense.

I'm now coming back to what I mentioned [here](https://github.com/snapshot-labs/snapshot/issues/887):

> even though the long term vision is to eradicate the need for libraries entirely as the browser-native APIs evolve, e.g. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat/format.

I'll give that a try now.

**Comment by samuveth** [comment]
> Looks good to me. Actually I don't really know why we didn't go for that right away, since we're building on the cutting edge anyway. I'll give that a try now.

Looks great. I don't see any reason not to use this. We will still use dayjs for all the other functionalities right?

**Comment by samuveth** [comment]
Should merge this without doing internationalisation and do that in another PR?

**Comment by samuveth** [comment]
@mktcode what your status on this?

**Comment by mktcode** [comment]
> @mktcode what your status on this?

\*sigh\* That's my status. :D I will try to get this to a conclusion this weekend.

**Comment by samuveth** [comment]
@mktcode I think we shouldn't bother to much with internationalisation, if there isn't an easy solution we should not spend to much time on this.

**Comment by samuveth** [comment]
https://github.com/snapshot-labs/snapshot/pull/1434

**Pull request closed without merging**